src/routes/safe/components/Balances/SendModal/screens/ContractInteraction/ReviewCustomTx/index.tsx

```diff
@@ -42,17 +44,18 @@ const ReviewCustomTx = ({ onClose, onPrev, tx }: Props): React.ReactElement => {
   const dispatch = useDispatch()
   const { address: safeAddress } = useSelector(safeSelector) || {}
   const [gasCosts, setGasCosts] = useState<string>('< 0.001')
-
+  const { nativeCoin } = getNetworkInfo()
   useEffect(() => {
     let isCurrent = true
 
     const estimateGas = async () => {
-      const { fromWei, toBN } = getWeb3().utils
       const txData = tx.data ? tx.data.trim() : ''
 
-      const estimatedGasCosts = await estimateTxGasCosts(safeAddress as string, tx.contractAddress as string, txData)
-      const gasCostsAsEth = fromWei(toBN(estimatedGasCosts), 'ether')
-      const formattedGasCosts = formatAmount(gasCostsAsEth)
+      const estimatedGasCosts = await (
+        await estimateTxGasCosts(safeAddress as string, tx.contractAddress as string, txData)
+      ).toString()
```

Comment:
Looking at all these `.toString()` calls, maybe we can add the `number` type to the `amount` parameter in `fromTokenUnit` function. As `BigNumber` can handle both types. WDYT?

Reply:
+1